grails-app/services/org/pih/warehouse/inventory/InventoryService.groovy

```diff
@@ -3432,4 +3436,22 @@ class InventoryService implements ApplicationContextAware {
             }
         }
     }
+
+    Boolean areLotsValid(List<Object> items) {
+        return items?.every {
+            InventoryItem inventoryItem = InventoryItem.findByLotNumberAndProduct(it?.lotNumber, Product.findByProductCode(it?.productCode))
+
+            if (!inventoryItem) {
+                return true
+            }
+
+            Integer availableQuantity = productAvailabilityService.getQuantityOnHand(inventoryItem)
+            Date expirationDate = it?.expirationDate != JSONObject.NULL ? Date.parse("MM/dd/yyyy", it?.expirationDate) : null
```

Comment:
I think that I saw this `JSONObject.NULL` causing some issues, but I am curious how we should handle this in Grails 1 in this case?

cc @kchelstowski @drodzewicz @awalkowiak